Zend/zend_compile.c

```diff
@@ -8902,6 +8902,10 @@ ZEND_API bool zend_binary_op_produces_error(uint32_t opcode, const zval *op1, co
 		/* Division by zero throws an error. */
 		return 1;
 	}
+	if ((opcode == ZEND_POW) && zval_get_long(op1) == 0 && zval_get_long(op2) < 0) {
+		/* 0 ** (<=0) throws a division by zero error. */
+		zend_error(E_DEPRECATED, "Power of base 0 and negative exponent is deprecated");
```

Comment:
This function shouldn't error itself. It should indicate whether the operation would error at runtime. In that case, it is not executed at compile time, so that the error may be caught at runtime. Hence, you can just return `true` here.